java-core/hash-map/hash-map.md

```diff
@@ -0,0 +1,19 @@
+### Common mistakes
```

Comment:
maybe add a reminder that they should be able to work with null as key, or it's covered ok in tests?

Reply:
we have a lot of tests for null key/value. And we added information about it in the theory, so I think it's enough